### PR TITLE
Migrate to UBI based go-build

### PR DESCRIPTION
**Pull request by hjiawei**
## Description

This PR contains the following changes:
* Bump calico/go-build to v0.90 which is based off RedHat UBI.
* 32 bit linux/arm platform is dropped because [Kubernetes also dropped it since v1.27](https://github.com/kubernetes/kubernetes/pull/115742).
* Update LLVM to v16 (from v15). The is the default in UBI 8.9.
* A few minor README and Makefile cleanups.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

N/A

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Comment by hjiawei** [comment]
Almost all green build [reference](https://tigera.semaphoreci.com/workflows/76c698fe-b78b-41ed-86dc-7d20db46b967?pipeline_id=e5a2b64f-3830-46b3-bda8-0e3c9f8314c1).

**Comment by hjiawei** [comment]
The "OpenStack integration (Ussuri)" test failure also happens [on master run](https://tigera.semaphoreci.com/workflows/cd52417e-0cf9-4366-8882-cfa71cdba0ac?pipeline_id=955fea12-a8b2-4718-9ceb-e72c5eb5b82c). It doesn't seem to be caused by this PR or the go-build changes.

File: metadata.mk [hunk 1]
```diff
@@ -3,7 +3,8 @@
 #################################################################################################
 
 # The version of github.com/projectcalico/go-build to use.
-GO_BUILD_VER = v0.89
+GO_BUILD_VER = ubi8
```
Comment (hjiawei) [inline metadata.mk:6]: - [ ] Revert once reviewed/validated.

File: felix/Makefile [hunk 2]
```diff
@@ -67,7 +67,7 @@ endif
 
 FV_ETCDIMAGE?=$(ETCD_IMAGE)
 FV_TYPHAIMAGE?=felix-test/typha:latest-$(BUILDARCH)
-FV_K8SIMAGE=calico/go-build:$(GO_BUILD_VER)
+FV_K8SIMAGE=jha48/go-build:$(GO_BUILD_VER)
```
Comment (hjiawei) [inline felix/Makefile:70]: - [ ] Revert once reviewed/validated.

File: lib.Makefile [hunk 3]
```diff
@@ -148,7 +130,7 @@ endif
 # For building, we use the go-build image for the *host* architecture, even if the target is different
 # the one for the host should contain all the necessary cross-compilation tools
 # we do not need to use the arch since go-build:v0.15 now is multi-arch manifest
-GO_BUILD_IMAGE ?= calico/go-build
+GO_BUILD_IMAGE ?= jha48/go-build
```
Comment (hjiawei) [inline lib.Makefile:133]: - [ ] Revert once reviewed/validated.

File: pod2daemon/Makefile [hunk 4]
```diff
@@ -107,10 +105,6 @@ endif
 else ifeq ($(ARCH), $(filter $(ARCH),arm64))
 CGO_ENABLED=0
 REGISTRAR_BUILD_CMD=$(REGISTRAR_TIGERA_BUILD_CMD)
-else ifeq ($(ARCH), $(filter $(ARCH),armv7))
```
[the remark this answers is not in the document]
Comment (hjiawei) [reply pod2daemon/Makefile:110]: That is correct. Kubernetes [dropped 32bit arm support](https://github.com/kubernetes/kubernetes/pull/115742) from v1.27+ and this is communicated/agreed with PM.

**Review by coutinhop — APPROVED** [review]

**Pull request merged**